### PR TITLE
[TRACES][OTEL] allow otel_debugging_on to emit traces to otel host

### LINKED ISSUE #43427: `otel_debugging_on` emits traces ONLY to the console and not to the configured otel host
### Apache Airflow version

2.10.2

### If "Other Airflow 2 version" selected, which one?

_No response_

### What happened?

I configured Airflow with the following OpenTelemetry Traces configuration - 

```
airflow:
  image:
    tag: 2.10.2-python3.12
  extraPipPackages:
    - "apache-airflow[otel]"
  config:
    AIRFLOW__TRACES__OTEL_DEBUGGING_ON: "True"
    AIRFLOW__TRACES__OTEL_ON: "True"
    AIRFLOW__TRACES__OTEL_HOST: "my_otel_host"
    AIRFLOW__TRACES__OTEL_PORT: <my_otel_port>
    AIRFLOW__TRACES__OTEL_SERVICE: "airflow"
```
    
  The above configuration when applied should emit airflow traces to the configured otel host. But they are only getting emitted in the console and not to the configured otel host.
  The reason for this is the `otel_debugging_on` attribute. When this is set to `true`, we see the above behaviour where the airflow does not emit traces to the host and only emits in the console. The [airflow docs](https://airflow.apache.org/docs/apache-airflow/stable/configurations-ref.html#config-traces-otel-debugging-on) also do mention that it will emit traces to both console and the otel host.
  If the attribute is removed from the otel traces config or is set to `False` we can see the traces emitted to the configured host.
  
  Proposed Change - 
  The `otel_debugging_on` when set to true in otel traces configuration, the traces should get emitted to both the console and the configured otel host.
  
  
 Airflow Version Used - `2.10.2`

### What you think should happen instead?

_No response_

### How to reproduce

Configure your airflow with the following traces configuration and check your otel host for traces. The traces should be visible with the config applied - 

```
    AIRFLOW__TRACES__OTEL_ON: "True"
    AIRFLOW__TRACES__OTEL_HOST: "my_otel_host"
    AIRFLOW__TRACES__OTEL_PORT: <my_otel_port>
    AIRFLOW__TRACES__OTEL_SERVICE: "airflow"
```

Now add the following attribute within the above configured traces config. Once this is added, the traces are no longer visible in the configured otel host but are visible in the console/logs.

```
    AIRFLOW__TRACES__OTEL_DEBUGGING_ON: "True"
```

### Operating System

Linux

### Versions of Apache Airflow Providers

_No response_

### Deployment

Other 3rd-party Helm chart

### Deployment details

Used [airflow community helm chart](https://github.com/airflow-helm/charts/tree/main/charts/airflow) for deploying airflow.

### Anything else?

_No response_

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by ninad-stackgen**
closes: #43427
related: #43427

### What? 
When traces are enabled via otel traces config, the traces will ALWAYS be emitted to the configured host via the `OTLPSpanExporter`. (even if `otel_debugging_on` is set to `true`)
If debug is enabled via `otel_debugging_on`, the traces will be emitted to the console as well (via the `ConsoleSpanExporter`).

### Why?
When otel traces are configured and `otel_debugging_on` is set to `True`, the traces are only logged to the console and not emitted to the configured host. The current change allows the traces to be emitted to both the console and the host when 
debug is enabled. 

### How? 
Removed the debug check from the get_otel_tracer() function. The function now always sets the span exporter as `OTLPSpanExporter`. Shifted the debug check in the get_tracer() function, where the `ConsoleSpanExporter` is added to the span exporter only when debug is true.

### CODE REVIEW

**Review by ferruzzi — APPROVED** [review]
Seems logical to me, I'll approve on the condition that @howardyoo agrees since he is much more familiar with the Traces stuff.

**Comment by ferruzzi** [comment]
This comment is here to block merging while it is open; feel free to resolve this after (and only after) @howardyoo gives his blessing.

File: airflow/traces/otel_tracer.py [hunk 1]
```diff
@@ -57,7 +57,7 @@ class OtelTrace:
     When OTEL is enabled, the Trace class will be replaced by this class.
```
Comment (ferruzzi) [inline airflow/traces/otel_tracer.py:57]: This comment is here to block merging while it is open; feel free to resolve this after (and only after) @howardyoo gives his blessing.

**Comment by howardyoo** [comment]
Hey all,
I would generally give my blessing to such a work. This would make
debugging OTEL instrumentation much easier and does not require extra
things (like running otel collector with debug exporter), so I would give a
thumbs up on this.

Thanks for the work!

**Review by potiuk — APPROVED** [review]

**Comment by ferruzzi** [comment]
Thanks @howardyoo, It seemed good to me but I wasn't involved in the Traces side like I was in the Metrics side so I wanted to make sure I wasn't missing some nuance.

**Comment by yatw** [comment]
@ninad-opsverse thank you for raising this issue. My team has been stuck on this for days until running into this post.
Could you kindly tell me how you trace the issue all the way to this part of the library? I want to become a dev that can troubleshoot problems in library too.

Also, is this fix available in the new version? My team is using 2.10.3 with this issue and I don't see this PR#43500 included in the release 2.10.5 or 2.10.4

**Pull request merged**